lib/src/main/scala/code/chymyst/jc/ReactionSite.scala

```diff
@@ -134,13 +132,13 @@ private[jc] final class ReactionSite(reactions: Seq[Reaction], reactionPool: Poo
     // Compute error messages here in case we will need them later.
     val blockingMoleculesWithNoReply = usedInputs
       .filter(_._2.reactionSentNoReply)
-      .keys.toSeq.toOptionSeq.map(_.map(_.toString).sorted.mkString(", "))
+      .map(_._1).toSeq.toOptionSeq.map(_.map(_.toString).sorted.mkString(", "))
```

Comment:
I will refactor them when I need to use more of these things. Tuples are useful for short pieces of code when they occur just a couple of times.